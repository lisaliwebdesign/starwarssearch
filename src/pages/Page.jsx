import React, { Component, Fragment } from "react";
import axios from "axios";
import { inject, observer } from "mobx-react";
import { Motion, spring } from "react-motion";
import Nav from "../components/Nav";
import PageItem from "../components/PageItem";
import {
  PageContainer //Set div tag for page
} from "../elements/page";

@inject("PageStore")
@observer
export default class Page extends React.Component {
  //Page onload
  componentDidMount() {
    let isSearch = !this.props.location.state;
    this.props.PageStore.loadData(
      isSearch,
      this.props.location.pathname,
      this.props.match.params.people
    );
  }
  //Update comopment when received new data
  componentWillReceiveProps(nextProps) {
    let isSearch = !nextProps.location.state;
    nextProps.PageStore.loadData(isSearch, nextProps.location.pathname);
  }

  render() {
    const PageStore = this.props.PageStore;
    if (PageStore.loadDataError) {
      return <PageContainer>Sorry... very embarassing</PageContainer>;
    }
    if (!PageStore.pageData) {
      return <PageContainer>Loading...</PageContainer>;
    }
    return (
      <PageContainer>
        <Nav
          name={
            PageStore.pageName +
            (PageStore.pageData.name
              ? PageStore.pageData.name
              : PageStore.pageData.title)
          }
        />
        <PageItem
          data={PageStore.pageData}
          onSelect={(path, location) => {
            this.props.history.push({
              pathname: path,
              state: location
            });
          }}
        />
      </PageContainer>
    );
  }
}
