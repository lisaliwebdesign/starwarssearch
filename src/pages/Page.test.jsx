import Page from "./Page";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

it("<pages><Page> - renders component", () => {
  const wrapper = render(<Page PageStore={{}} />);
  expect(wrapper).toMatchSnapshot();
});

it("<pages><Page> - renders loading", () => {
  const wrapper = render(<Page PageStore={{}} />);
  //console.log(wrapper.html());
  expect(wrapper.html()).toEqual("Loading...");
});

it("<pages><Page> - renders error", () => {
  const wrapper = render(<Page PageStore={{ loadDataError: true }} />);
  expect(wrapper.html()).toEqual("Sorry... very embarassing");
});

it("<pages><Page> - renders with data", () => {
  //Mock router
  const mockRouter = {
    childContextTypes: {
      router: () => void 0
    },
    context: {
      router: {
        history: "test",
        route: {
          location: {
            hash: "",
            pathname: "",
            search: "",
            state: ""
          },
          match: { params: {}, isExact: false, path: "", url: "" }
        }
      }
    }
  };
  // mount component with router context and get component's instance
  const wrapper = shallow(
    <Page
      Nav={<div>testnav</div>}
      PageStore={{
        pageName: "testpagename",
        pageData: { name: "testpagename" }
      }}
    />,
    mockRouter
  );
  // console.log(wrapper.props(), wrapper.state());
  expect(wrapper).toMatchSnapshot();
});
