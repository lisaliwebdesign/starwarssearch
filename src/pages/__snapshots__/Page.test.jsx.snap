// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<pages><Page> - renders component 1`] = `
<div
  class="sc-ifAKCX fKJMqe"
>
  Loading...
</div>
`;

exports[`<pages><Page> - renders with data 1`] = `
<Page
  Nav={
    <div>
      testnav
    </div>
  }
  PageStore={
    Object {
      "pageData": Object {
        "name": "testpagename",
      },
      "pageName": "testpagename",
    }
  }
/>
`;
